Bump some old deps to avoid breakage and hence allow allow other dep
issues to be debugged with -Z minimal-versions

encoding bumped because version 0.2.0 of encoding crate is broken
flate2 bumped because version 0.3.0 of gcc crate is broken
bumped because version 0.5.0 of proc-macro-hack is broken
syntect and thiserror bumped to make syntect build.

Index: bat/Cargo.toml
===================================================================
--- bat.orig/Cargo.toml
+++ bat/Cargo.toml
@@ -63,7 +63,7 @@ version = "0.15.5"
 version = "0.2.4"
 
 [dependencies.encoding_rs]
-version = "0.8.33"
+version = "0.8"
 
 [dependencies.etcetera]
 version = "0.8.0"
@@ -110,7 +110,7 @@ optional = true
 version = "1.0"
 
 [dependencies.serde]
-version = "1.0"
+version = "1.0.142"
 features = ["derive"]
 
 [dependencies.serde_yaml]
@@ -126,7 +126,7 @@ features = ["parsing"]
 default-features = false
 
 [dependencies.thiserror]
-version = "1.0"
+version = "1.0.30"
 
 [dependencies.unicode-width]
 version = "0.1.10"
